File: android/app/src/main/java/com/adblock/Statistics.kt
package com.adblock

/**
 * Statistics data class for tracking ad blocking metrics
 */
data class Statistics(
    val blockedCount: Int,
    val allowedCount: Int,
    val dataSaved: Int
) {
    val blockRate: Double
        get() {
            val total = blockedCount + allowedCount
            return if (total > 0) (blockedCount.toDouble() / total) * 100 else 0.0
        }
}

File: ios/AdBlock/Statistics.swift
import Foundation

/// Statistics for ad blocking activity
public struct Statistics: Codable {
    /// Number of blocked requests
    public let blockedCount: Int
    
    /// Number of allowed requests
    public let allowedCount: Int
    
    /// Estimated data saved in bytes
    public let dataSaved: Int
    
    /// Initialize statistics
    public init(blockedCount: Int, allowedCount: Int, dataSaved: Int) {
        self.blockedCount = blockedCount
        self.allowedCount = allowedCount
        self.dataSaved = dataSaved
    }
    
    /// Total number of requests processed
    public var totalRequests: Int {
        return blockedCount + allowedCount
    }
    
    /// Block rate as a percentage (0-100)
    public var blockRate: Double {
        guard totalRequests > 0 else { return 0 }
        return Double(blockedCount) / Double(totalRequests) * 100
    }
    
    /// Human-readable data saved
    public var dataSavedFormatted: String {
        let formatter = ByteCountFormatter()
        formatter.countStyle = .binary
        return formatter.string(fromByteCount: Int64(dataSaved))
    }
}